libcaf_net/test/net/web_socket/framing.cpp

```diff
@@ -162,4 +163,205 @@ SCENARIO("the client closes the connection with a closing handshake") {
+                                      input, 0);
+      transport->push(input);
+      input.clear();
+      detail::rfc6455::assemble_frame(detail::rfc6455::ping, 0x0, data, input);
+      transport->push(input);
+      input.clear();
+      data = as_bytes(make_span(fragment2));
+      detail::rfc6455::assemble_frame(detail::rfc6455::continuation_frame, 0x0,
+                                      data, input);
+      transport->push(input);
+      transport->handle_input();
+      THEN("the server responds with a pong") {
+        detail::rfc6455::header hdr;
+        auto hdr_len
+          = detail::rfc6455::decode_header(transport->output_buffer(), hdr);
+        MESSAGE("Payload: " << transport->output_buffer());
+        CHECK_EQ(hdr_len, 2u);
+        CHECK(hdr.fin);
+        CHECK_EQ(hdr.opcode, detail::rfc6455::pong);
+        CHECK_EQ(hdr.payload_len, 5u);
+        CHECK_EQ(hdr.mask_key, 0u);
+      }
+      THEN("the server receives the full text message") {
+        CHECK_EQ(app->text_input, "Hello, world!"sv);
+      }
+      AND("the client did not abort") {
+        CHECK(!app->has_aborted());
+      }
+    }
+  }
+}
+
+SCENARIO("the client sends an fragmented text message with a ping in-between "
+         "separated by octets") {
+  GIVEN("a valid WebSocket connection") {
+    WHEN("the client sends the first text frame, a ping, and then the final "
+         "text frame separately") {
+      auto fragment1 = "Hello"sv;
+      auto fragment2 = ", world!"sv;
+      std::vector<std::byte> input;
+      auto data = as_bytes(make_span(fragment1));
+      detail::rfc6455::assemble_frame(detail::rfc6455::text_frame, 0x0, data,
+                                      input, 0);
+      transport->push(input);
+      transport->handle_input();
+      THEN("the server receives nothing") {
+        CHECK(app->text_input.empty());
+        CHECK(app->binary_input.empty());
+      }
+      input.clear();
+      detail::rfc6455::assemble_frame(detail::rfc6455::ping, 0x0, data, input);
+      transport->push(input);
+      transport->handle_input();
+      THEN("the server responds with a pong") {
+        detail::rfc6455::header hdr;
+        auto hdr_len
+          = detail::rfc6455::decode_header(transport->output_buffer(), hdr);
+        CHECK_EQ(hdr_len, 2u);
+        CHECK(hdr.fin);
+        CHECK_EQ(hdr.opcode, detail::rfc6455::pong);
+        CHECK_EQ(hdr.payload_len, 5u);
+        CHECK_EQ(hdr.mask_key, 0u);
+      }
+      input.clear();
+      data = as_bytes(make_span(fragment2));
+      detail::rfc6455::assemble_frame(detail::rfc6455::continuation_frame, 0x0,
+                                      data, input);
+      transport->push(input);
+      transport->handle_input();
+      THEN("the server receives the full text message") {
+        CHECK_EQ(app->text_input, "Hello, world!"sv);
+      }
+      AND("the client did not abort") {
+        CHECK(!app->has_aborted());
+      }
+    }
+  }
+}
+
 END_FIXTURE_SCOPE()
+
```

Comment:
All tests end up in the same binary, so we should put local classes into the anonymous namespace.

```suggestion

namespace {

```